src/NUnitTestAdapter/NUnitTestAdapter.cs

```diff
@@ -81,16 +56,51 @@ public static bool IsRunningUnderIDE
 
         #region Protected Helper Methods
 
+        // The Adapter is constructed using the default constructor.
+        // We don't have any info to initialize it until one of the
+        // ITestDiscovery or ITestExecutor methods is called. The
+        // each Discover or Execute method must call this method.
+        protected virtual void Initialize(IMessageLogger messageLogger)
+        {
+            _verbosity = 0; // In case we throw below
+            _adapterVersion = Assembly.GetExecutingAssembly().GetName().Version.ToString();
+
+            try
+            {
+                var registry = RegistryCurrentUser.OpenRegistryCurrentUser(@"Software\nunit.org\VSAdapter");
+                UseVsKeepEngineRunning = registry.Exist("UseVsKeepEngineRunning") && (registry.Read<int>("UseVsKeepEngineRunning") == 1);
+                _shadowCopy = registry.Exist("ShadowCopy") && (registry.Read<int>("ShadowCopy") == 1);
```

Comment:
The shadow copy should be false by default, and yes, we discussed that earlier with that conclusion. 
I did the registry as a an temporary solution.  It should be settings controlled through the Tools/Options in VS, but where the actual storing should be is still open.  
The runsettings-file can be extended, but very few uses it, so forcing people to add it just for the settings doesn't seem right.  VS do have a lot of settings in the registry, so that is still a viable option, the adapter is anyway part of VS.